### PR TITLE
[Bug] Skip WebAuthn 2fa event logs during login flow

### PR DESCRIPTION
**Pull request by vincentsalucci**
## Type of change
- [X] Bug fix
- [ ] New feature development
- [ ] Tech debt (refactoring, code cleanup, dependency upgrades, etc)
- [ ] Build/deploy pipeline (DevOps)
- [ ] Other

## Code changes
- **WebAuthnTokenProvider.cs**: Updated methods to pass parameter for skipping log events 
- **UserService.cs**: Updated method to handle default boolean for logging user event for updating/enabling 2fa

## Testing requirements
- Make sure changes properly suppress the event logs with a WebAuthn 2fa login flow

## Before you submit
- [X] I have checked for formatting errors (`dotnet tool run dotnet-format --check`) (required)
- [ ] If making database changes - I have also updated Entity Framework queries and/or migrations
- [ ] I have added **unit tests** where it makes sense to do so (encouraged but not required)
- [ ] This change requires a **documentation update** (notify the documentation team)
- [ ] This change has particular **deployment requirements** (notify the DevOps team)

### CODE REVIEW

File: src/Core/Identity/WebAuthnTokenProvider.cs [hunk 1]
```diff
@@ -77,7 +77,7 @@ public async Task<string> GenerateAsync(string purpose, UserManager<User> manage
             var providers = user.GetTwoFactorProviders();
             providers[TwoFactorProviderType.WebAuthn] = provider;
             user.SetTwoFactorProviders(providers);
-            await userService.UpdateTwoFactorProviderAsync(user, TwoFactorProviderType.WebAuthn);
+            await userService.UpdateTwoFactorProviderAsync(user, TwoFactorProviderType.WebAuthn, true, false);
```
Comment (Hinton) [inline src/Core/Identity/WebAuthnTokenProvider.cs:80]: I think the following pattern is easier to read.
```suggestion
            await userService.UpdateTwoFactorProviderAsync(user, TwoFactorProviderType.WebAuthn, logEvent: true);
```

File: src/Core/Identity/WebAuthnTokenProvider.cs [hunk 2]
```diff
@@ -123,7 +123,7 @@ public async Task<bool> ValidateAsync(string purpose, string token, UserManager<
             var providers = user.GetTwoFactorProviders();
             providers[TwoFactorProviderType.WebAuthn].MetaData[webAuthCred.Item1] = webAuthCred.Item2;
             user.SetTwoFactorProviders(providers);
-            await userService.UpdateTwoFactorProviderAsync(user, TwoFactorProviderType.WebAuthn);
+            await userService.UpdateTwoFactorProviderAsync(user, TwoFactorProviderType.WebAuthn, true, false);
```
Comment (Hinton) [inline src/Core/Identity/WebAuthnTokenProvider.cs:126]:
```suggestion
            await userService.UpdateTwoFactorProviderAsync(user, TwoFactorProviderType.WebAuthn, logEvent: true);
```

**Review by Hinton — CHANGES_REQUESTED** [review]

File: src/Core/Identity/WebAuthnTokenProvider.cs (hunk 1 above)
Comment (vincentsalucci) [reply src/Core/Identity/WebAuthnTokenProvider.cs]: Neeeeeat. Updated!

File: src/Core/Identity/WebAuthnTokenProvider.cs (hunk 2 above)
Comment (vincentsalucci) [reply src/Core/Identity/WebAuthnTokenProvider.cs]: 👍

**Review by Hinton — APPROVED** [review]

**Pull request merged**